migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,295 @@
+		updatedSchema.NotifiersTableName)
+	row, err := database.PostgresDB.Query(database.DBCtx, sql, notifiertID)
+	if err != nil {
+		return false, err
+	}
+	if row.Next() {
+		return true, nil
+	}
+	return false, nil
+}
+
+func getMigratedReportConfigIfExsists(reportID string, database *types.Databases) (bool, *storage.ReportConfiguration, error) {
+	sql := fmt.Sprintf(
+		"SELECT serialized FROM %s WHERE id = $1",
+		updatedSchema.ReportConfigurationsTableName)
+	row, err := database.PostgresDB.Query(database.DBCtx, sql, getDeterministicID(reportID))
+	if err != nil {
+		return false, nil, err
+	}
+	if row.Next() {
+		var data []byte
+		if err := row.Scan(&data); err != nil {
+			return false, nil, err
+		}
+		var config storage.ReportConfiguration
+		if err := config.Unmarshal(data); err != nil {
+			return false, nil, err
+		}
+		return true, &config, nil
+	}
+	return false, nil, nil
+}
+
+func migrate(database *types.Databases) error {
+
+	db := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt)
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportSnapshotsStmt)
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt.Children[0])
+	db = db.WithContext(database.DBCtx)
+	tx := db.Begin()
+	query := db.WithContext(database.DBCtx).Table(updatedSchema.ReportConfigurationsTableName).Select("serialized")
+	rows, err := query.Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to iterate table %s", "report_configurations")
+	}
+	defer func() { _ = rows.Close() }()
+	var convertedReportConfigs []*updatedSchema.ReportConfigurations
+	var reportSnapshots []*updatedSchema.ReportSnapshots
+	var reportNotifiers []*updatedSchema.ReportConfigurationsNotifiers
+	var errList errorhelpers.ErrorList
+	for rows.Next() {
+		// convert to report config proto
+		var reportConfig *updatedSchema.ReportConfigurations
+		if err = query.ScanRows(rows, &reportConfig); err != nil {
+			return errors.Wrapf(err, "failed to scan rows")
+		}
+		reportConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert report config  %+v to proto", reportConfigProto)
+		}
+		// skip if version=0 and scope id is not nil since it is v2 config created in tech preview
+		if reportConfigProto.Version == 0 && reportConfigProto.GetResourceScope() != nil {
+			reportConfigProto.Version = 2
+			// convert report config proto back to gorm model
+			convertedGormConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert report config from proto %+v", reportConfigProto)
+			}
+			convertedReportConfigs = append(convertedReportConfigs, convertedGormConfig)
+
+			continue
+		}
+
+		//if deterministic id exists no need to copy the config
```

Comment:
What are we trying to do here?  Are we trying to catch the case of a previous migration or the case where one of the batches of this migration created the config?